monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/python
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for check_collectd_mlab check."""
+
+import os
+import socket
+import time
+import unittest2 as unittest
+
+# Third-party modules.
+import mock
+
+# Module under test.
+import check_collectd_mlab
+
+# R0904: Too many public methods. Hard to avoid for unit tests.
+# pylint: disable=R0904
+
+
+class MLabNagiosSocketTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+
+  def testunit_sock_sendcmd_RETURNS_successfully(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = list('1 default reply\n')
```

Comment:
My preference is to use `.return_value` in cases like this when we're always returning a single value because `side_effect` yields unintuitive behavior when we're trying to do something simple. Like here, where I believed this code causes our mocked responses to `recv` to each be 1 character (see my other comment below) even though the code makes it looks like it's returning a string.

Are we deliberately using `.side_effect` because we *want* to simulate behavior where the first call yields the initial answer and subsequent calls fail?

My first choice here would be to just use `return_value` because I don't think it's important to verify `recv` was only called once, but I'm also okay with:

```
mock_sock.recv.side_effect = ('1 default reply\n',)
```